shared/src/balancer_event_handler.rs

```diff
@@ -93,31 +302,38 @@ pub struct BalancerEventUpdater(
 );
 
 impl BalancerEventUpdater {
-    pub async fn new(contract: BalancerV2Vault, pools: BalancerPools) -> Result<Self> {
+    pub async fn new(contract: BalancerV2Vault, pools: BalancerPools) -> Self {
         let deployment_block = match contract.deployment_information() {
             Some(DeploymentInformation::BlockNumber(block_number)) => Some(block_number),
-            Some(DeploymentInformation::TransactionHash(hash)) => Some(
-                contract
+            Some(DeploymentInformation::TransactionHash(hash)) => {
+                match contract
                     .raw_instance()
                     .web3()
                     .block_number_from_tx_hash(hash)
-                    .await?,
```

Comment:
Is this change because of xDAI? We could also make this continue erroring and then configure the xDAI nodes to not specify `--baseline-liquidiy Balancer`.

Reply:
I am not sure I recall what this was for. Will have to look into this one.

Reply:
Actually. I think this was the result of a bad merge.